akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpRequestParser.scala

```diff
@@ -167,6 +167,8 @@ private[http] class HttpRequestParser(_settings: ParserSettings,
             emitRequestStart(emptyEntity(cth))
             setCompletionHandling(HttpMessageParser.CompletionOk)
             startNewMessage(input, bodyStart)
+          } else if (!method.isEntityAccepted) {
+            failMessageStart(UnprocessableEntity, s"${method.name} requests must not have an entity")
```

Comment:
No, I'd vote to keep it there.
The `require` in the `HttpRequest` makes sure that we find problems like the one fixed here.
Also, it protects users from creating illegal requests in the first place.